src/librustdoc/lib.rs

```diff
@@ -258,7 +279,18 @@ pub fn main_args(args: &[String]) -> isize {
     let css_file_extension = matches.opt_str("e").map(|s| PathBuf::from(&s));
     let cfgs = matches.opt_strs("cfg");
 
+    // we now check if unstable options are allowed and if we're in a nightly build
+    let has_z_unstable_options = matches.opt_strs("Z")
+                                        .iter()
+                                        .any(|x| *x == "unstable-options");
```

Comment:
It looks like this will accept anything after `-Z`. I think there needs to be a branch returning an error if any z-flags are passed besides `unstable_options`.